### DISCUSSION TITLE
Regarding annotations and permalinks

### DISCUSSION THREAD

**Discussion by AmritasyaPutra** [idno/idno#3073] [Q&A]
I just migrated my WordPress site to withknown. I am very happy with it. I noticed some deadlinks on my withknown site now. They are annotations. The permalink at the time of import was /:year/:month/:slug  and later I changed it to /:year/:month/:slug which seems to have led to this problem.

For example, the annotation on the following article does not have the 'day' before the slug, hence it is a dead link.
https://k.ekvastra.in/2016/06/doctrine-of-karma---law-of-action-and-reaction/annotations/2bb3586c30604de77d3fe54b42c62536

https://k.ekvastra.in/2016/06/13/doctrine-of-karma---law-of-action-and-reaction

I know that this is embedded in JSON so I won't attempt to edit the DB. Is there a simple way to fix this?

**Comment by thierrymarianne** [comment]
By targeting the columns of the record matching the aforementioned entry, 
JSON and substitution functions could be used to add the missing URL part.

Otherwise manually editing the JSON field if needed could also be a viable option with an editor helping in checking the JSON document syntax validity. 
Depending on the database behind your website, it could more or less doable without much trouble.

⚠ Of course, I'd highly recommend you make a backup of your database 
before doing anything in any case 😅
